pkgs/development/libraries/opencolorio/1.x.nix

```
@@ -33,7 +33,12 @@ stdenv.mkDerivation rec {
     ++ lib.optional (stdenv.isDarwin && stdenv.isAarch64) "-DCMAKE_OSX_ARCHITECTURES=arm64";
 
   postInstall = ''
-    mkdir -p $bin/bin; mv $out/bin $bin/
+    moveToOutput bin "$bin"
+    moveToOutput cmake "$dev"
+    moveToOutput OpenColorIOConfig.cmake "$dev"
+
+    substituteInPlace "$dev/cmake/OpenColorIO-release.cmake" \
+      --replace "$out/bin" "$bin/bin"
```

```suggestion
    substituteInPlace OpenColorIO-release.cmake \
      --replace "$out/bin" "$bin/bin"

    moveToOutput OpenColorIOConfig.cmake "$dev"
```
so we don't need to know the path of the cmake file

---

Also when I locally checked out the PR and build it as is the cmake file was in $dev/OpenColorIOConfig.cmake

---

Did you _try_ this? Doesn't seem to me like it will work, `substituteInPlace` not being able to find `OpenColorIO-release.cmake`. Note there are a number of cmake files in the `cmake` subdir, only `cmake/OpenColorIO-release.cmake` needing substitutions.

---

:shrug:

---

> Did you _try_ this?

The suggestion? I never try those due to lack of time. If I try them I might as well PR them.

The cmake file is still in the wrong place when checking the PR out as is.

```
 ➜ nix build opencolorio_1.dev

 ➜ tree result-dev
 result-dev ⇒ /nix/store/q859n8jgg7a3lgj2gzwzwd1wnn8jwwdi-opencolorio-1.1.1-dev
├──  cmake
│   ├──  OpenColorIO-release.cmake
│   └──  OpenColorIO.cmake
├──  include
│   └──  OpenColorIO
│       ├──  OpenColorABI.h
│       ├──  OpenColorIO.h
│       ├──  OpenColorTransforms.h
│       └──  OpenColorTypes.h
├──  lib
│   └──  pkgconfig
│       └──  OpenColorIO.pc
├──  nix-support
│   └──  propagated-build-inputs
└──  OpenColorIOConfig.cmake
```

---

#209550